gpflow/posterior.py

```diff
@@ -346,8 +346,15 @@ def _get_Kff(self, Xnew, full_cov):
     def _conditional_with_precompute(
         self, Xnew, full_cov: bool = False, full_output_cov: bool = False
     ) -> MeanAndVariance:
-        # Qinv: [L, M, M]
+        # Qinv: [M, M] or [L, M, M]
         # alpha: [M, L]
+        if self.Qinv.shape.ndims == 2:
+            # When using a shared kernel and shared inducing points, the shape of Qinv is [M, M].
+            # We need to manually reintroduce the axis representing the number of latent GPs.
+            L = self.q_dist.q_mu.shape[1]
+            Qinv = tf.repeat(self.Qinv[None, :], L, axis=0)
```

Comment:
> Are you suggesting using `broadcast_to` instead of `repeat` here?

That would probably be better already, yeah. I'm just surprised it doesn't broadcast automatically!

Reply:
i.e., why is
```suggestion
            Qinv = self.Qinv[None, :]
```
not enough? :thinking:

Reply:
The combination of `IndependentPosteriorMultiOutput` with a `SharedIndependent` kernel and `SharedIndependentInducingVariables` means that there is no record of the number of latent GPs when the covariance is computed. The mean is fine because `alpha` still retains an `L` axis.

I wonder if ensuring the `Qinv` always has an `L` axis would also fix the other problem with the shape of the covariance?

Reply:
I'd be happy to always have an L axis:)